UiPath.Ipc.net6.0-windows.received.txt

```diff
@@ -0,0 +1,264 @@
+        public UiPath.Ipc.ClientTransport Transport { get; init; }
+        public TProxy GetProxy<TProxy>()
+            where TProxy :  class { }
+    }
+    public class IpcProxy : System.Reflection.DispatchProxy, System.IDisposable
+    {
+        public IpcProxy() { }
+        public System.IO.Stream? Network { get; }
+        public event System.EventHandler ConnectionClosed;
+        public System.Threading.Tasks.ValueTask CloseConnection() { }
+        public void Dispose() { }
+        protected override object? Invoke(System.Reflection.MethodInfo? targetMethod, object?[]? args) { }
+    }
+    public sealed class IpcServer : System.IAsyncDisposable
+    {
+        [System.Obsolete("Constructors of types with required members are not supported in this version of " +
+            "your compiler.", true)]
+        public IpcServer() { }
+        public UiPath.Ipc.EndpointCollection Endpoints { get; init; }
+        public System.Collections.Generic.IReadOnlyList<UiPath.Ipc.ListenerConfig> Listeners { get; init; }
+        public System.Threading.Tasks.TaskScheduler? Scheduler { get; init; }
+        public System.IServiceProvider ServiceProvider { get; init; }
+        public System.Threading.Tasks.ValueTask DisposeAsync() { }
+        public void Start() { }
+        public System.Threading.Tasks.Task WaitForStart() { }
+        public System.Threading.Tasks.Task WaitForStop() { }
+    }
+    public abstract class ListenerConfig : UiPath.Ipc.EndpointConfig, System.IEquatable<UiPath.Ipc.ListenerConfig>
+    {
+        protected ListenerConfig() { }
+        public System.Security.Cryptography.X509Certificates.X509Certificate? Certificate { get; init; }
+        public int ConcurrentAccepts { get; init; }
+        public byte MaxReceivedMessageSizeInMegabytes { get; init; }
+    }
+    public class Message
+    {
+        public Message() { }
+        [Newtonsoft.Json.JsonIgnore]
+        public UiPath.Ipc.IClient Client { get; set; }
+        [Newtonsoft.Json.JsonIgnore]
+        public System.TimeSpan RequestTimeout { get; set; }
+        public TCallbackInterface GetCallback<TCallbackInterface>()
+            where TCallbackInterface :  class { }
+        public void ImpersonateClient(System.Action action) { }
+    }
+    public class Message<TPayload> : UiPath.Ipc.Message
+    {
+        public Message(TPayload payload) { }
+        public TPayload Payload { get; }
+    }
+    [System.Serializable]
+    public class RemoteException : System.Exception
+    {
+        public RemoteException(UiPath.Ipc.Error error) { }
+        public UiPath.Ipc.RemoteException? InnerException { get; }
+        public override string StackTrace { get; }
+        public string Type { get; }
+        public bool Is<TException>()
+            where TException : System.Exception { }
+        public override string ToString() { }
+    }
+}
+namespace UiPath.Ipc.Extensibility
+{
+    public interface IListenerConfig<TSelf, TListenerState, TConnectionState>
```

Comment:
IListenerConfig vs ListenerConfig not obvious.
Why other namespace not obvious.
Maybe you'd want to move all transport stuff into UiPath.Ipc.Transport (client included), but I don't see the need.

Too many generics, one of the problems we wanted to avoid.